src/Symfony/Component/Routing/Loader/ObjectLoader.php

```diff
@@ -52,7 +52,7 @@ public function load($resource, $type = null)
         $loaderObject = $this->getObject($parts[0]);
 
         if (!\is_object($loaderObject)) {
-            throw new \LogicException(sprintf('%s:getObject() must return an object: %s returned', \get_class($this), \gettype($loaderObject)));
+            throw new \TypeError(sprintf('%s:getObject() must return an object: %s returned', \get_class($this), \gettype($loaderObject)));
```

Comment:
This change mimics the behavior we would get after adding return types. Once we add return type declarations, this whole `if` block could be removed.

Reply:
That was actually a bad idea. I've marked the corresponding test as legacy instead.